docs/guidelines/git-signed-commits.md

```diff
@@ -0,0 +1,172 @@
+| **Cross-platform Support** | Broad support across various platforms and Git tools.        | Well-supported, with widespread integration in Git tools.      | Support varies; some tools may not support S/MIME directly.   |
+| **Pros**                   | - Simplifies workflow by using the same keys for authentication and signing. <br> - Integrated into SSH, which is commonly used for secure Git operations. | - Decentralized and flexible, with a variety of algorithms and key sizes. <br> - Well-established in the open-source community. | - Trust model is straightforward, based on established CAs. <br> - May align with existing certificate-based security practices (e.g., email). |
+| **Cons**                   | - Primarily verifies the commit was pushed by an authenticated user, not necessarily the commit's author. <br> - SSH key management is crucial; compromised keys pose a risk. | - Key management and the web of trust model can be complex. <br> - Requires active key maintenance (revocation, expiration). | - Dependent on third-party CAs for issuance and trust. <br> - Certificates have expiration dates and may incur costs. |
+
+!!! Note Signing method
+
+    We recommend using self-signed SSH keys for signing your git commits (In the future we may switch to a certificate based approach)
+
+## Configure your local development environment
+
+For this guideline our reference is using Git from the command line. 
+
+We assume that Git and SSH is installed on your system. 
+
+We assume that you are using SSH to authenticate Git with github.com. Consult our [guideline](git-github.md) for more information on this topic.
+
+The Github documentation on [SSH commit signature verification](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification#ssh-commit-signature-verification) is a good source for detailed information.
+
+### Adding a SSH key for signing
+
+!!! Note Use multiple SSH keys
+
+    We recommend using different SSH keys for authentication and signing. This may add some extra complexity and it may provide a more robust set-up with looser coupling of key components of the SDLC.
+
+To create a new SSH key for signing you can use the following command (alter the date manually):
+
+```shell
+ssh-keygen -t ed25519 -f ~/.ssh/git_ssh_signing_key_1 -C "Created on <date>, for larskaare on github.com"
+```
+
+This will create a SSH signing key and add a comment on date and purpose. Add a passphrase to the key. Success full key generation will output the key fingerprint and a randomart image (randomart is supposed to be a visualisation making it easier to validate keys - and identify changes)
+
+Add the new key to the ssh-agent
+
+```shell
+ssh-add ~/.ssh/git_ssh_signing_key_1
```

Comment:
I suggest that we also include the tip about adding the passphrase to the keychain:
For MacOS we would typically add the passphrase to the keychain as well
```shell
ssh-add --apple-use-keychain ~/.ssh/github_larskaare_1
```

Maybe for windows users suggest an alternative?

Reply:
The tip on passphrase is mentioned quite a few times above - so i guess that will do. The example of using the keychain or similar is mentioned in the git guideline. I'll link that one.